{
  "Severity": "HIGH",
  "UpdatedAt": "2020-08-31T19:01:07Z",
  "Package": {
    "Ecosystem": "NPM",
    "Name": "http-proxy"
  },
  "Advisory": {
    "DatabaseId": 3040,
    "Id": "MDE2OlNlY3VyaXR5QWR2aXNvcnlHSFNBLTZ4MzMtcHc3cC1obXBx",
    "GhsaId": "GHSA-6x33-pw7p-hmpq",
    "References": [
      {
        "Url": "https://github.com/http-party/node-http-proxy/pull/1447/files"
      },
      {
        "Url": "https://github.com/advisories/GHSA-6x33-pw7p-hmpq"
      }
    ],
    "Identifiers": [
      {
        "Type": "GHSA",
        "Value": "GHSA-6x33-pw7p-hmpq"
      }
    ],
    "Description": "Versions of `http-proxy` prior to 1.18.1 are vulnerable to Denial of Service. An HTTP request with a long body triggers an `ERR_HTTP_HEADERS_SENT` unhandled exception that crashes the proxy server. This is only possible when the proxy server sets headers in the proxy request using the `proxyReq.setHeader` function.   \n\nFor a proxy server running on `http://localhost:3000`, the following curl request triggers the unhandled exception:  \n```curl -XPOST http://localhost:3000 -d \"$(python -c 'print(\"x\"*1025)')\"```\n\n\n## Recommendation\n\nUpgrade to version 1.18.1 or later",
    "Origin": "UNSPECIFIED",
    "PublishedAt": "2020-09-04T17:59:49Z",
    "Severity": "HIGH",
    "Summary": "Denial of Service in http-proxy",
    "UpdatedAt": "2020-09-04T17:59:49Z",
    "WithdrawnAt": ""
  },
  "Versions": [
    {
      "FirstPatchedVersion": {
        "Identifier": "1.18.1"
      },
      "VulnerableVersionRange": "\u003c 1.18.1"
    }
  ]
}